src/use_promise_guard/index.es

```diff
@@ -0,0 +1,19 @@
+import useMounted from '../use_mounted';
+
+const usePromiseGuard = (deps) => {
+  const isMounted = useMounted(deps);
```

Comment:
I think it doesn't make sense to useMounted with any kind of deps at the moment. 

Lets say you pass some id as deps: [id]
Initially, react calls the effect and sets isMoutned.current to true,
then the id changes to some other value.
That causes React to set isMounted.current to false and then re-execute the effect and set isMounted.current to true.

The time window in which isMounted.current is set to false is a very short one. The only situation when it's false is when the component unmounted.